pinot-common/src/main/java/org/apache/pinot/common/config/provider/TableCache.java

```diff
@@ -361,7 +361,8 @@ private void putSchema(ZNRecord znRecord)
         columnNameMap.put(columnName, columnName);
       }
     }
-    _schemaInfoMap.put(schemaName, new SchemaInfo(schema, columnNameMap));
+    _schemaInfoMap.put(schemaName, new SchemaInfo(schema, columnNameMap, schema.getDimensionFieldSpecs().size(),
+        schema.getDateTimeFieldSpecs().size(), schema.getMetricFieldSpecs().size()));
```

Comment:
Can we derive these fields from the schema object rather than passing them explicitly to `SchemaInfo` constructor?

Reply:
Will check it out.